core/src/main/java/org/apache/gravitino/listener/api/info/partitions/PartitionInfo.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.gravitino.listener.api.info.partitions;
+
+import com.google.common.collect.ImmutableMap;
+import java.util.Map;
+import org.apache.gravitino.annotation.DeveloperApi;
+import org.apache.gravitino.rel.partitions.IdentityPartition;
+import org.apache.gravitino.rel.partitions.ListPartition;
+import org.apache.gravitino.rel.partitions.Partition;
+import org.apache.gravitino.rel.partitions.RangePartition;
+
+/**
+ * Provides access to metadata about a Partition instance, designed for use by event listeners. This
+ * class encapsulates the essential attributes of a Partition, including its name and properties
+ * information.
+ */
+@DeveloperApi
+public class PartitionInfo {
+  private String name;
+  private Map<String, String> properties;
+
+  /**
+   * Constructs PartitionInfo from an existing Partition object.
+   *
+   * @param partition The Partition instance to extract information from.
+   */
+  public PartitionInfo(Partition partition) {
+    this(partition.name(), partition.properties());
+  }
+
+  /**
+   * Directly constructs PartitionInfo with specified details.
+   *
+   * @param name The name of the Partition.
+   * @param properties A map of properties associated with the Partition.
+   */
+  public PartitionInfo(String name, Map<String, String> properties) {
```

Comment:
make it private

Reply:
the method cannot be made private, because it is still needed by child classes

Reply:
If declare the PartitionInfo class as an abstract class, this will cause other tests to fail because PartitionEventDispatcher needs to construct PartitionInfo with an existing partition.

So I think the best way is that **declare this constructor as a protected method**, which can meet our requirements

Reply:
simple `PartitionInfo` instance is meaningless to me.